src/lib.rs

```diff
@@ -73,10 +73,21 @@ fn merge_results(result: &mut CovResult, result2: &mut CovResult) {
     }
 }
 
-fn add_results(mut results: Vec<(String,CovResult)>, result_map: &SyncCovResultMap) {
+fn add_results(mut results: Vec<(String,CovResult)>, result_map: &SyncCovResultMap, source_dir: &Option<PathBuf>) {
     let mut map = result_map.lock().unwrap();
     for mut result in results.drain(..) {
-        match map.entry(result.0) {
+        let path = if source_dir.is_some() {
+            // the goal here is to be able to merge results for paths like foo/./bar and foo/bar
+            let source_dir = source_dir.clone().unwrap();
```

Comment:
If you use match instead of if is_some maybe you can avoid cloning.

```Rust
let path = match source_dir {
    Some(source_dir) => {
      // ...
    },
    None => result.0
}
```